Src/CSharpier.Cli/HasMismatchedCliAndMsBuildVersions.cs

```diff
@@ -21,7 +21,20 @@ public static bool Check(string directory, IFileSystem fileSystem, ILogger logge
         {
             // this could potentially use the Microsoft.CodeAnalysis.Project class, but that was
             // proving difficult to use
-            var csProjXElement = XElement.Load(fileSystem.File.OpenRead(pathToCsProj));
+            XElement csProjXElement;
```

Comment:
Odd. How come this doesn't need to be an `XElement?`?

Reply:
I missed the `?` next to XElement and was very confused by your question at first...

After playing around it seems to be because if you do something like this
```c#
XElement csProjXElement;
var x = csProjXElement.XPathSelectElements(xpath).FirstOrDefault();
```
Then you get the error
```
Local variable 'csProjXElement' might not be initialized before accessing
```

So it's possible to have non-nullable element that isn't immediately initialized.